src/service/locator.cpp

```diff
@@ -714,20 +733,20 @@ locator_t::on_context_shutdown() {
             COCAINE_LOG_DEBUG(m_log, "shutting down %d remote client(s)", mapping.size());
         }
 
-        mapping.clear();
+        boost::for_each(mapping | boost::adaptors::map_values, [](client_map_t::mapped_type& ptr) {
+            ptr->detach(std::error_code{});
+        });
     });
 
     m_remotes.apply([this](remote_map_t& mapping) {
-        m_cluster = nullptr;
-
         if(mapping.empty()) {
             return;
         } else {
-            COCAINE_LOG_DEBUG(m_log, "closing %d outgoing locator streams", mapping.size());
+            COCAINE_LOG_DEBUG(m_log, "closing %d outgoing locator stream(s)", mapping.size());
         }
 
         boost::for_each(mapping | boost::adaptors::map_values, [](streamed<results::connect>& s) {
-            try { s.close(); } catch(...) { /* None */ }
+            try { s.close(); } catch(...) { /***/ }
```

Comment:
What do you want to happen here then? Spam logs with "we tried to close a stream on core shutdown but it was already closed"?